### PR TITLE
Init repo, port registry from dss-charter

### PR DESCRIPTION
**Pull request by talbaneth**

### CODE REVIEW

**Comment by talbaneth** [comment]
Note for reviewers - the contract and test where already reviewed in dss-charter repo.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# CDP Registry
+![Build Status](https://github.com/makerdao/dss-cdp-registry/actions/workflows/.github/workflows/tests.yaml/badge.svg?branch=master)
+
+This contract is used to reserve a CDP id from a CDP manager.  
+It restrics a user from registering more than one CDP id per collateral type.
```
Comment (kmbarry1) [inline README.md]: restrics --> restricts

File: src/CdpRegistry.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    constructor(address cdpManager_) public {
+        cdpManager = cdpManager_;
+    }
+
+    function open(
+        bytes32 ilk,
+        address usr
+    ) public returns (uint256) {
+        require(usr == msg.sender, "usr-not-sender");
```
Comment (kmbarry1) [inline src/CdpRegistry.sol:40]: Why even have `usr`? Just a back compat thing?

Comment (talbaneth) [reply src/CdpRegistry.sol:40]: Yes, the idea was to have the same interface as dss-cdp-manager's open().

File: README.md (hunk 1 above)
Comment (talbaneth) [reply README.md]: thanks, fixed

**Comment by kmbarry1** [comment]
Hmmmm looks like the CI is running into some sort of error.

**Comment by gbalabasquer** [comment]
> Hmmmm looks like the CI is running into some sort of error.

Fixed.

**Review by hexonaut — APPROVED** [review]

**Review by gbalabasquer — APPROVED** [review]
LGTM

**Pull request merged**